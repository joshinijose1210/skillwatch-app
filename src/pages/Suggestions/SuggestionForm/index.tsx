export { SuggestionForm as default } from './SuggestionForm';
