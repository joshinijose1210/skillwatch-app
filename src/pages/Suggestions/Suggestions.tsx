import { PageContent } from '@components';
import { CustomTable } from '@components/reusableComponents/CustomTable/CustomTable';
import ListHeader from '@components/reusableComponents/ListHeader';
import { Tabs, Button } from '@medly-components/core';
import { CallMadeIcon, CallReceivedIcon } from '@medly-components/icons';
import DOMPurify from 'dompurify';
import { SubmittedCols, ReceivedCols } from './columns';
import {
    StyledTableWrapper,
    StyledTabs,
    StyledModalContentFlex,
    SuggestionLayer,
    SuggestionDescriptionLayer,
    StyledHeading,
    StyledText,
    StyledHTMLText,
    StyledPopup,
    FilterContainer,
    ProgressFilterWrapper,
    ProgressAndCommentWrapper,
    SuggestionCommentContainer,
    ProgressWrapper,
    CommentWrapper
} from './Suggestions.style';
import { useSuggestion } from './useSuggestion';
import { StyledModalTitle, StyledSingleSelect, SuggestionProgressId } from '@common';
import TooltipDropdown from '@components/reusableComponents/TooltipDropdown';
import { OptionsType } from '@components/reusableComponents/TooltipDropdown/types';
import { StyledModalHeader, StyledModalContent, StyledModalActions } from '@common';
import { RichTextEditor } from '@components/reusableComponents/RichTextEditor'; // Adjust path as needed

export const Suggestions = () => {
    const {
        modalState,
        suggestedDate,
        suggestedBy,
        suggestionText,
        isReceivedAllowed,
        openModal,
        closeModal,
        addSuggestionPage,
        handleTabChange,
        suggestionsData,
        totalSuggestionsDataCount,
        handlePageChange,
        page,
        activeTab,
        suggestionsDataIsLoading,
        handleSortChange,
        tagsList,
        handleDropdownChange,
        progressStatus,
        handleProgressSubmission,
        progressLoading,
        suggestionReceivedFilter,
        suggestionFilterOptions,
        handleSuggestionReceievedDropdownChange,
        suggestionPendingCount,
        suggestionComment,
        handlesuggestionCommentChange,
        comments,
        removeHtmlTags
    } = useSuggestion();

    const MIN_COMMENT_LENGTH = 10;
    const MAX_COMMENT_LENGTH = 100;

    const rawSuggestionComment = removeHtmlTags(suggestionComment).trim();

    const isSaveDisabled =
        !progressStatus ||
        (progressStatus === SuggestionProgressId.Deferred && !rawSuggestionComment) ||
        (rawSuggestionComment.length > 0 && rawSuggestionComment.length < MIN_COMMENT_LENGTH) ||
        rawSuggestionComment.length > MAX_COMMENT_LENGTH;

    return (
        <PageContent>
            <ListHeader title="Suggestions" actionButtonLabel="Add Suggestion" actionButtonClick={addSuggestionPage} />

            <StyledPopup open={modalState} onCloseModal={closeModal}>
                <StyledModalHeader>
                    <StyledModalTitle textVariant="h2">{`Suggestion`}</StyledModalTitle>
                </StyledModalHeader>
                <StyledModalContent>
                    <StyledModalContentFlex>
                        <SuggestionLayer>
                            <StyledHeading textWeight="Medium">
                                Date: <StyledText>{suggestedDate}</StyledText>
                            </StyledHeading>
                        </SuggestionLayer>
                        <SuggestionLayer>
                            <StyledHeading textVariant="h4" textWeight="Medium">
                                Suggested By: <StyledText>{suggestedBy}</StyledText>
                            </StyledHeading>
                        </SuggestionLayer>
                        <SuggestionDescriptionLayer>
                            <StyledHeading textVariant="h4" textWeight="Medium">
                                Suggestion:
                            </StyledHeading>
                            <StyledHTMLText dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(suggestionText) }} />
                        </SuggestionDescriptionLayer>
                        {activeTab === 'receivedSuggestion' && (
                            <ProgressAndCommentWrapper>
                                <ProgressWrapper>
                                    <StyledSingleSelect
                                        options={tagsList}
                                        variant="outlined"
                                        placeholder="Select Progress"
                                        size="M"
                                        label="Progress"
                                        value={progressStatus}
                                        isSearchable
                                        minWidth="100%"
                                        onChange={val => val && handleDropdownChange(val)}
                                        data-testid="progress"
                                    />
                                    <TooltipDropdown
                                        dataIds={[`progress-input`]}
                                        values={
                                            progressStatus && tagsList.length
                                                ? tagsList.filter((item: OptionsType) => progressStatus === item.value)
                                                : []
                                        }
                                    />
                                </ProgressWrapper>
                                <CommentWrapper>
                                    <RichTextEditor
                                        fullWidth
                                        editor="suggestion-comment"
                                        input={suggestionComment}
                                        placeholder="Add comment here..."
                                        setInput={handlesuggestionCommentChange}
                                        height={100}
                                        minLength={MIN_COMMENT_LENGTH}
                                        maxLength={MAX_COMMENT_LENGTH}
                                        marginBottom={2}
                                    />
                                </CommentWrapper>
                            </ProgressAndCommentWrapper>
                        )}
                        {comments.length > 0 && (
                            <SuggestionDescriptionLayer>
                                {comments.map(c => (
                                    <SuggestionCommentContainer key={c.id}>
                                        <StyledHeading textVariant="h4" textWeight="Medium">
                                            Comment: <StyledText>{c.date}</StyledText>
                                        </StyledHeading>
                                        <StyledHTMLText dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(c.comment) }} />
                                    </SuggestionCommentContainer>
                                ))}
                            </SuggestionDescriptionLayer>
                        )}
                    </StyledModalContentFlex>
                </StyledModalContent>
                {activeTab === 'receivedSuggestion' && (
                    <StyledModalActions>
                        <Button isLoading={progressLoading} disabled={isSaveDisabled} onClick={handleProgressSubmission}>
                            Save
                        </Button>
                    </StyledModalActions>
                )}
            </StyledPopup>
            <FilterContainer>
                <ProgressFilterWrapper>
                    <StyledSingleSelect
                        options={suggestionFilterOptions}
                        variant="outlined"
                        placeholder="Select Progress"
                        data-testid="progressDropdown"
                        label={'Progress'}
                        onChange={e => handleSuggestionReceievedDropdownChange(e)}
                        value={suggestionReceivedFilter.progressId}
                        size="M"
                        minWidth="25rem"
                    />
                </ProgressFilterWrapper>
            </FilterContainer>
            <StyledTabs aria-label="Closed style tabs" tabSize="M" variant="outlined" onChange={id => handleTabChange(id)}>
                <Tabs.Tab
                    active={activeTab === 'submittedSuggestion'}
                    id="submittedSuggestion"
                    label="Submitted"
                    helperText="Submitted"
                    icon={CallMadeIcon}
                    data-testid="submitted"
                />
                <Tabs.Tab
                    active={activeTab === 'receivedSuggestion'}
                    id="receivedSuggestion"
                    label="Received"
                    helperText="Received"
                    icon={CallReceivedIcon}
                    count={suggestionPendingCount}
                    hide={!isReceivedAllowed}
                    data-testid="received"
                />
            </StyledTabs>
            <StyledTableWrapper>
                <CustomTable
                    data={suggestionsData}
                    tableKey={activeTab}
                    columns={activeTab === 'submittedSuggestion' ? SubmittedCols : ReceivedCols}
                    isLoading={suggestionsDataIsLoading}
                    activePage={page || 1}
                    count={totalSuggestionsDataCount || 0}
                    handlePageChange={handlePageChange}
                    setSortOrder={handleSortChange}
                />
            </StyledTableWrapper>
        </PageContent>
    );
};
