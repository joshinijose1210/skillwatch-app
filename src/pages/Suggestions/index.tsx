export { Suggestions as default } from './Suggestions';
