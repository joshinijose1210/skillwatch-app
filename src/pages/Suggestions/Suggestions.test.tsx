import MockAdapter from 'axios-mock-adapter';
import { apiInstance } from '@utils';
import { Suggestions } from './Suggestions';
import { routeConstants } from '@constants';
import { RTKRender, cleanup, fireEvent, screen, waitFor } from '@utils/test-utils';
import { notificationCountData, suggestionMockData } from '@mocks/suggestions';
import * as reactRouterDom from 'react-router-dom';
import { userMockState } from '@mocks/userMockState';

const mock = new MockAdapter(apiInstance, { onNoMatch: 'throwException' });

const mockedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedNavigate,
    useLocation: jest.fn()
}));

const useLocationMock = jest.spyOn(reactRouterDom, 'useLocation');
let mockedState = { ...suggestionMockData.mockState, action: '' };
const originalScrollIntoWindow = window.HTMLElement.prototype.scrollIntoView;

beforeEach(() => {
    useLocationMock.mockReturnValue({
        ...suggestionMockData.locationMockData,
        pathname: `${routeConstants.suggestionBox}/1`,
        state: mockedState
    });
    const { url, progressListUrl, progressList, responseCode, data } = suggestionMockData;
    mock.onGet(url).replyOnce(responseCode, data);
    mock.onGet(progressListUrl).replyOnce(responseCode, progressList);
    window.HTMLElement.prototype.scrollIntoView = jest.fn();
});

afterEach(() => {
    cleanup();
    mock.reset();
    window.HTMLElement.prototype.scrollIntoView = originalScrollIntoWindow;
});

describe('Suggestions component', () => {
    it('should take suggestion component snapshot', () => {
        const { container } = RTKRender(<Suggestions />);
        expect(container).toMatchSnapshot();
    });

    it('should sorting work propely', async () => {
        const { url, data } = suggestionMockData;
        mock.onGet(url).replyOnce(200, data);
        RTKRender(<Suggestions />);
        await waitFor(() => screen.findAllByText('View'), { timeout: 3000 });
        const spanElement = screen.getByText('Date');
        fireEvent.click(spanElement);
        await waitFor(() => expect(screen.getAllByText('View')[0]).toBeInTheDocument(), { timeout: 3000 });
    });

    it('should open and close view suggestion popup when clicked on `View` Action', async () => {
        mockedState = { ...suggestionMockData.mockState, action: 'View' };
        useLocationMock.mockReturnValue({
            ...suggestionMockData.locationMockData,
            pathname: `${routeConstants.suggestionBox}/1`,
            state: mockedState
        });
        const { url, data } = suggestionMockData;
        mock.onGet(url).replyOnce(200, data);
        RTKRender(<Suggestions />);
        await waitFor(() => screen.findAllByText('View'), { timeout: 3000 });
        fireEvent.click(screen.getAllByText('View')[0]);
        expect(await screen.findByText('Test User (C0001)')).toBeInTheDocument();
        fireEvent.click(screen.getByTitle('medly-modal-close-icon'));
        await waitFor(() => expect(screen.queryByText('Test User (C0001)')).not.toBeInTheDocument(), { timeout: 3000 });
    });

    it('should add suggestion button rendered properly and must be clickable', async () => {
        const { url, data } = suggestionMockData;
        mock.onGet(url).replyOnce(200, data);
        RTKRender(<Suggestions />, {
            initialEntries: [`${routeConstants.suggestionBox}/1`]
        });
        const button = screen.getByRole('button', { name: 'Add Suggestion' });
        fireEvent.click(button);
        expect(mockedNavigate).toHaveBeenCalledWith(`${routeConstants.suggestionBox}/undefined/add-suggestion`, {
            state: { action: 'Add' }
        });
    });

    it('should progress update work properly', async () => {
        const { receivedUrl, progressUpdateUrl, responseCode, data } = suggestionMockData;
        const { url: notificationURL, data: notificationData, notPendingData } = notificationCountData;
        mock.onGet(receivedUrl).replyOnce(200, data);
        mock.onGet(notificationURL).replyOnce(200, notificationData);
        mock.onPatch(progressUpdateUrl).replyOnce(responseCode, {});
        mock.onGet(notificationURL).replyOnce(200, notPendingData);
        RTKRender(<Suggestions />, {
            initialState: userMockState
        });
        fireEvent.click(screen.getByTestId('received'));
        await waitFor(() => screen.findAllByText('View'), { timeout: 3000 });
        expect(screen.getByText('2')).toBeInTheDocument();
        fireEvent.click(screen.getAllByText('View')[0]);
        fireEvent.change(screen.getByTestId('progress'), { target: { value: 'Completed' } });
        fireEvent.click(screen.getByText('Save'));
        expect(screen.getByText('1')).toBeInTheDocument();
        expect(await screen.findByText('Suggestion progress updated successfully')).toBeInTheDocument();
    });

    it('should check if progress update API error is handled properly', async () => {
        mockedState = { ...suggestionMockData.mockState, suggestedByFirstName: '' };
        useLocationMock.mockReturnValue({
            ...suggestionMockData.locationMockData,
            pathname: `${routeConstants.suggestionBox}/1`,
            state: mockedState
        });
        const { receivedUrl, progressUpdateUrl, data } = suggestionMockData;
        mock.onGet(receivedUrl).replyOnce(200, data);
        mock.onPatch(progressUpdateUrl).replyOnce(404, {
            errorMessage: 'something went wrong'
        });
        RTKRender(<Suggestions />, {
            initialState: userMockState
        });
        fireEvent.click(screen.getByTestId('received'));
        await waitFor(() => screen.findAllByText('View'), { timeout: 3000 });
        fireEvent.click(screen.getAllByText('View')[0]);
        fireEvent.change(screen.getByTestId('progress'), { target: { value: 'Completed' } });
        fireEvent.click(screen.getByText('Save'));
        expect(await screen.findByText('something went wrong')).toBeInTheDocument();
    });

    it('should check if filter works properly', async () => {
        const { progressFilterAllUrl, progressFilterAllData } = suggestionMockData;
        const { progressFilterUrl, progressFilterPendingData } = suggestionMockData;

        mock.onGet(progressFilterAllUrl).replyOnce(200, progressFilterAllData);
        mock.onGet(progressFilterUrl).replyOnce(200, progressFilterPendingData);

        RTKRender(<Suggestions />, {
            initialState: userMockState
        });
        fireEvent.click(screen.getByTestId('received'));

        await waitFor(() => screen.findAllByText('View'), { timeout: 3000 });

        expect(screen.getAllByText('Completed').length).toEqual(3);
        const progressDropdown = screen.getByTestId('progressDropdown');

        expect(progressDropdown).toBeInTheDocument();
        fireEvent.click(progressDropdown);

        const PendingStatus = await screen.findAllByText('Pending');
        fireEvent.click(PendingStatus[0]);

        expect((await screen.findAllByText('Pending')).length).toEqual(progressFilterPendingData.totalSuggestions);
    });
});
