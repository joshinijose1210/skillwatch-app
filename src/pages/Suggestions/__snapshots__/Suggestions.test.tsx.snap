// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestions component should take suggestion component snapshot 1`] = `
<div>
  <main
    class="sc-1ehavxz-0 dDyhBJ"
  >
    <div
      class="sc-h70dbj-0 fQmYvl row expanded space-between align-center marginBottom-2"
    >
      <div
        class="sc-w8e7z6-2"
      >
        <h3
          class="sc-1ntgl3l-0 kcLBOi sc-w8e7z6-0 ibnmt"
          textalign="initial"
          textcolor="inherit"
          textvariant="h3"
          textweight="Medium"
        >
          Suggestions
        </h3>
      </div>
      <button
        class="sc-1k2druj-0 eLgfbc sc-w8e7z6-1"
        edges="square"
        type="button"
        variant="solid"
      >
        <span
          class="sc-1ntgl3l-0 eOsHHD"
          textalign="initial"
          textcolor="inherit"
          textvariant="button1"
        >
          Add Suggestion
        </span>
      </button>
    </div>
    <div
      class="sc-1etsj2e-0 gMXChf sc-tkiih6-20 sc-1e4bh26-6 iSFEzl ldgATE"
      id="medly-modal"
      open=""
    >
      <div
        class="sc-bu1pt3-0 kLJgCq"
        id="medly-modal-popup"
        open=""
      >
        <svg
          class="sc-1pf3x8a-0 eHKFnt sc-1jyrp7a-0 gPLpLo"
          fill="none"
          height="1em"
          id="medly-modal-close-button"
          title="medly-modal-close-icon"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.3 5.71a.996.996 0 0 0-1.41 0L12 10.59 7.11 5.7A.996.996 0 1 0 5.7 7.11L10.59 12 5.7 16.89a.996.996 0 1 0 1.41 1.41L12 13.41l4.89 4.89a.996.996 0 1 0 1.41-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
            fill="#000"
            fill-opacity=".54"
          />
        </svg>
        <div
          class="sc-1etsj2e-1 jRtjKC"
          headerheight="0"
          id="medly-modal-inner-container"
        >
          <div
            class="sc-1whgp8r-0 cmHeGg"
            id="medly-modal-header"
            scrollstate="[object Object]"
          >
            <h2
              class="sc-1ntgl3l-0 hJdvRC sc-tkiih6-27 ebKlKN"
              textalign="initial"
              textcolor="inherit"
              textvariant="h2"
            >
              Suggestion
            </h2>
          </div>
          <div
            class="sc-1dfqrlx-0 bmLVFj sc-tkiih6-25"
            headerheight="0"
            id="medly-modal-content"
            scrollstate="[object Object]"
          >
            <div
              class="sc-1e4bh26-5 dOXwqB"
            >
              <div
                class="sc-1e4bh26-7 kRrAsF"
              >
                <span
                  class="sc-1ntgl3l-0 gPMgWC sc-1e4bh26-9 hghKjn"
                  textalign="initial"
                  textcolor="inherit"
                  textweight="Medium"
                >
                  Date: 
                  <span
                    class="sc-1ntgl3l-0 hGagSU sc-1e4bh26-10 bnjtMX"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    10/05/2024
                  </span>
                </span>
              </div>
              <div
                class="sc-1e4bh26-7 kRrAsF"
              >
                <h4
                  class="sc-1ntgl3l-0 jPeJxZ sc-1e4bh26-9 hghKjn"
                  textalign="initial"
                  textcolor="inherit"
                  textvariant="h4"
                  textweight="Medium"
                >
                  Suggested By: 
                  <span
                    class="sc-1ntgl3l-0 hGagSU sc-1e4bh26-10 bnjtMX"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    Test User (C0001)
                  </span>
                </h4>
              </div>
              <div
                class="sc-1e4bh26-8 iEYemz"
              >
                <h4
                  class="sc-1ntgl3l-0 jPeJxZ sc-1e4bh26-9 hghKjn"
                  textalign="initial"
                  textcolor="inherit"
                  textvariant="h4"
                  textweight="Medium"
                >
                  Suggestion:
                </h4>
                <span
                  class="sc-1ntgl3l-0 hGagSU sc-1e4bh26-11 dwWqKY"
                  textalign="initial"
                  textcolor="inherit"
                >
                  <p>
                    gfhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh
                  </p>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-1e4bh26-12 hxjtsI"
    />
    <div
      aria-label="Closed style tabs"
      class="sc-13bppbd-0 sc-tkiih6-28 sc-1e4bh26-0 dErHOx bPAAmU"
      id="medly-tabs"
    >
      <div
        class="sc-12mpb33-0 dYQjJT"
        id="medly-tabs-list"
        tabsize="M"
        variant="outlined"
      >
        <button
          aria-controls="panel-submittedSuggestion"
          aria-selected="true"
          bgcolor="[object Object]"
          bordercolor="[object Object]"
          class="sc-1kcg570-4 ewhzXK"
          countbgcolor="[object Object]"
          countborderradius="2.5rem"
          countcolor="#ffffff"
          data-testid="submitted"
          helpertextcolor="[object Object]"
          iconbgcolor="[object Object]"
          iconcolor="[object Object]"
          id="submittedSuggestion"
          label="[object Object]"
          labelcolor="[object Object]"
          role="tab"
          solid="[object Object]"
          tabbackground="WHITE"
          tabindex="0"
          tabsize="M"
          totaltabs="2"
          type="button"
          variant="outlined"
        >
          <svg
            class="sc-1pf3x8a-0 kdcvez"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.296 5.704c0 .56.45 1 1 1h5.59l-10.89 10.89a.996.996 0 1 0 1.41 1.41l10.89-10.89v5.59c0 .55.45 1 1 1s1-.45 1-1v-8c0-.55-.45-1-1-1h-8c-.55 0-1 .45-1 1z"
              fill="#000"
              fill-opacity=".54"
            />
          </svg>
          <div
            class="sc-1kcg570-5 kFiIrt"
          >
            <div
              class="sc-1kcg570-6 jobpmu"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1kcg570-1 jILKEJ"
                id="submittedSuggestion-label"
                tabsize="M"
                textalign="initial"
                textcolor="inherit"
              >
                Submitted
              </span>
            </div>
          </div>
        </button>
        <button
          aria-controls="panel-receivedSuggestion"
          aria-selected="false"
          bgcolor="[object Object]"
          bordercolor="[object Object]"
          class="sc-1kcg570-4 fpIKmc"
          countbgcolor="[object Object]"
          countborderradius="2.5rem"
          countcolor="#ffffff"
          data-testid="received"
          helpertextcolor="[object Object]"
          iconbgcolor="[object Object]"
          iconcolor="[object Object]"
          id="receivedSuggestion"
          label="[object Object]"
          labelcolor="[object Object]"
          role="tab"
          solid="[object Object]"
          tabbackground="WHITE"
          tabindex="-1"
          tabsize="M"
          totaltabs="2"
          type="button"
          variant="outlined"
        >
          <svg
            class="sc-1pf3x8a-0 kdcvez"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.008 5.001a.996.996 0 0 0-1.41 0l-10.89 10.88v-5.59c0-.55-.45-1-1-1s-1 .45-1 1v8c0 .55.45 1 1 1h8c.55 0 1-.45 1-1s-.45-1-1-1h-5.59l10.89-10.89c.38-.38.38-1.02 0-1.4z"
              fill="#000"
              fill-opacity=".54"
            />
          </svg>
          <div
            class="sc-1kcg570-5 kFiIrt"
          >
            <div
              class="sc-1kcg570-6 jobpmu"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1kcg570-1 jILKEJ"
                id="receivedSuggestion-label"
                tabsize="M"
                textalign="initial"
                textcolor="inherit"
              >
                Received
              </span>
            </div>
          </div>
        </button>
      </div>
      <div
        aria-labelledby="submittedSuggestion"
        class="sc-x0gfjw-0 bLZBSc"
        id="medly-tabs-panel-submittedSuggestion"
        role="tabpanel"
        tabindex="0"
      />
    </div>
    <div
      class="sc-1e4bh26-1 ijSwRe"
    >
      <div
        class="sc-8tbhpn-3"
      >
        <div
          class="sc-1msmxxr-0 iWEzjW"
        />
        <table
          actions=""
          class="sc-1msmxxr-1 jwFsPF sc-8tbhpn-2 kCDmJI"
          columns="[object Object],[object Object],[object Object],[object Object],[object Object]"
          defaultactivepage="1"
          defaultsortorder="desc"
          itemsperpage="10"
          rowclickdisablekey=""
          tabindex="-1"
          totalitems="0"
        >
          <thead
            class="sc-uxzsxp-0 haNltq"
          >
            <tr
              class="sc-1jr91bg-0 hJzMcO"
              gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 2fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
              style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 2fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
            >
              <th
                class="sc-hjmm4m-0 iOyNiA"
                tablesize="M"
              >
                <button
                  class="sc-hjmm4m-2 kpjakb"
                  tablesize="M"
                >
                  <span
                    class="sc-1ntgl3l-0 ghpJwn"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    Date
                  </span>
                  <svg
                    class="sc-1pf3x8a-0 bMQNVa"
                    fill="none"
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.229 6.935a1 1 0 0 1 1.543 0l2.208 2.679a1 1 0 0 1-.77 1.636H9.79a1 1 0 0 1-.77-1.636l2.209-2.679zM12.771 17.065a1 1 0 0 1-1.543 0L9.02 14.386a1 1 0 0 1 .772-1.636h4.418a1 1 0 0 1 .771 1.636l-2.209 2.679z"
                      fill="#435465"
                    />
                  </svg>
                </button>
                <span
                  class="sc-hjmm4m-1 cuEDpa"
                />
              </th>
              <th
                class="sc-hjmm4m-0 iOyNiA"
                tablesize="M"
              >
                <button
                  class="sc-hjmm4m-2 kpjakb"
                  tablesize="M"
                >
                  <span
                    class="sc-1ntgl3l-0 ghpJwn"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    Suggestion
                  </span>
                </button>
                <span
                  class="sc-hjmm4m-1 cuEDpa"
                />
              </th>
              <th
                class="sc-hjmm4m-0 iOyNiA"
                tablesize="M"
              >
                <button
                  class="sc-hjmm4m-2 kpjakb"
                  tablesize="M"
                >
                  <span
                    class="sc-1ntgl3l-0 ghpJwn"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    Status
                  </span>
                </button>
                <span
                  class="sc-hjmm4m-1 cuEDpa"
                />
              </th>
              <th
                class="sc-hjmm4m-0 iOyNiA"
                tablesize="M"
              >
                <button
                  class="sc-hjmm4m-2 kpjakb"
                  tablesize="M"
                >
                  <span
                    class="sc-1ntgl3l-0 ghpJwn"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    Progress
                  </span>
                </button>
                <span
                  class="sc-hjmm4m-1 cuEDpa"
                />
              </th>
              <th
                class="sc-hjmm4m-0 iOyNiA"
                tablesize="M"
              >
                <button
                  class="sc-hjmm4m-2 kpjakb"
                  tablesize="M"
                >
                  <span
                    class="sc-1ntgl3l-0 ghpJwn"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    Action
                  </span>
                </button>
                <span
                  class="sc-hjmm4m-1 cuEDpa"
                />
              </th>
            </tr>
          </thead>
          <tbody
            class="sc-c7cniy-0 kShdfA"
          >
            <tr
              class="sc-1yxujaa-0 bgyIwA"
              gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 2fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
              style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 2fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
              tabindex="0"
            >
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
            </tr>
            <tr
              class="sc-1yxujaa-0 bgyIwA"
              gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 2fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
              style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 2fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
              tabindex="1"
            >
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
            </tr>
            <tr
              class="sc-1yxujaa-0 bgyIwA"
              gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 2fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
              style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 2fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
              tabindex="2"
            >
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
            </tr>
            <tr
              class="sc-1yxujaa-0 bgyIwA"
              gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 2fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
              style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 2fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
              tabindex="3"
            >
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
            </tr>
            <tr
              class="sc-1yxujaa-0 bgyIwA"
              gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 2fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
              style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 2fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
              tabindex="4"
            >
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </main>
</div>
`;
